src/visualisation/panel.cc

```diff
@@ -230,97 +241,127 @@ void VisualisationPanel::refreshVisualisation() {
 void VisualisationPanel::initLayout() {
   initOptionsPanel();
 
-  QWidget* root_widget = new QWidget(this);
-  this->setCentralWidget(root_widget);
-
-  splitter_ = new QSplitter(Qt::Horizontal);
-  splitter_->addWidget(minimap_);
-  splitter_->addWidget(visualisation_);
-  splitter_->setStretchFactor(1, 5);
-
-  layout_ = new QHBoxLayout;
-  layout_->setContentsMargins(0, 0, 0, 0);
-  layout_->addWidget(splitter_, 8);
-  layout_->addLayout(options_layout_, 0);
-  root_widget->setLayout(layout_);
+  setCentralWidget(visualisation_root_);
+  setDockNestingEnabled(true);
+
+  node_tree_dock_ = new QDockWidget;
```

Comment:
Hmm, who takes ownership and when of this widget? When I saw similar code before I thought it was done by `addDockWidget`, however the documentation doesn't mention anything about it.